lib/galaxy/model/migrate/versions/0155_add_content_update_time.py

```diff
@@ -0,0 +1,105 @@
+"""
+Adds trigger to dataset table to update history.update_time when
+contents are changed.
+"""
+from __future__ import print_function
+
+import logging
+
+from sqlalchemy import DDL, MetaData
+
+log = logging.getLogger(__name__)
+metadata = MetaData()
+
+
+def upgrade(migrate_engine):
+    print(__doc__)
+    metadata.bind = migrate_engine
+    metadata.reflect()
+    drop_update_trigger(migrate_engine)
+    install_update_trigger(migrate_engine)
+
+
+def downgrade(migrate_engine):
+    metadata.bind = migrate_engine
+    metadata.reflect()
+    drop_update_trigger(migrate_engine)
+
+
+def not_pg(self, target, bind, state, **kw):
+    return bind.engine.name != 'postgresql'
+
+
+def install_update_trigger(migrate_engine):
+    """Installs trigger on database table to update history table
+    when contents have changed. Installs a function and a trigger
+    for postgres, other sql variants only require the trigger def
+    """
+
+    pg_create_trigger = DDL("""
+        CREATE FUNCTION update_history_content_update_time()
+            RETURNS trigger
+            LANGUAGE 'plpgsql'
+        AS $BODY$
+        begin
+            update history h
+            set update_time = current_timestamp
+            from history_dataset_association hda
+            where h.id = hda.history_id
+            and hda.dataset_id = NEW.id;
+            return NEW;
+        end;
+        $BODY$;
+        CREATE TRIGGER update_history_update_time
+            BEFORE INSERT OR DELETE OR UPDATE
```

Comment:
```suggestion
            AFTER INSERT OR DELETE OR UPDATE
```
See below for explanation.